CRM/Event/Form/Registration/Confirm.php

```diff
@@ -493,6 +493,28 @@ public function postProcess() {
         }
       }
 
+      // CRM-18397: Display billing address on contact summary page when values for all billing fields.
+      $addressFields = array(
+        "billing_street_address-{$this->_bltID}",
+        "billing_city-{$this->_bltID}",
+        "billing_state_province-{$this->_bltID}",
+        "billing_state_province_id-{$this->_bltID}",
+        "billing_postal_code-{$this->_bltID}",
+        "billing_country-{$this->_bltID}",
+      );
+
+      $billingAddressFields = array_intersect_key($value, array_flip($addressFields));
+      if (empty(array_filter($billingAddressFields))) {
+        foreach ($addressFields as $key) {
+          unset($value[$key]);
+        }
+        unset($value['billing_first_name']);
+        unset($value['billing_middle_name']);
+        unset($value['billing_last_name']);
+        unset($value["billing_country_id-{$this->_bltID}"]);
+        unset($value["address_name-{$this->_bltID}"]);
+      }
+
```

Comment:
I also have a bit of a vendetta against 'lists of fields' in Civi that are manually typed out like this. They are quite hard to maintain. I'll take another look at this in the alternate PR.